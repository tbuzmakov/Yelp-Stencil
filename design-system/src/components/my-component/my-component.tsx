import { Component, ComponentInterface, h, Host, Prop } from '@stencil/core';

@Component({
  tag: 'my-component',
  styleUrl: 'my-component.scss',
  shadow: true,
})
export class MyComponent implements ComponentInterface {
  @Prop() first?: string;
  @Prop() last?: string;

  render() {
    // return <Host>{`Hello!, my name is ${this.first} ${this.last}`}</Host>;
    return <Host>
      
      <div>
        <h1>Test</h1>
      </div>
      {`Hello!, my name is ${this.first} ${this.last}`}


    </Host>;
  }
}
