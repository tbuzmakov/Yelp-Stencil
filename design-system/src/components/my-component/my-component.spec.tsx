import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';

import { MyComponent } from './my-component';

describe('my component', () => {
  const getSpecPage = async (props: Omit<MyComponent, 'render'> = {}) =>
    await newSpecPage({
      components: [MyComponent],
      template: () => <my-component {...props} />,
    });

  it('should display the full name if passed', async () => {
    const page = await getSpecPage({ first: 'John', last: 'Doe' });

    expect(page.root).toMatchSnapshot();
  });
});
