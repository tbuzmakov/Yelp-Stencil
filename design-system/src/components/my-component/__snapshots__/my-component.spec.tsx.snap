// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my component should display the full name if passed 1`] = `
<my-component>
  <mock:shadow-root>
    Hello, my name is John Doe
  </mock:shadow-root>
</my-component>
`;
